### DISCUSSION TITLE
single_diode_model 30 min PV output

### DISCUSSION THREAD

**Discussion by AnnaGo12** [pvlib/pvlib-python#1639] [Q&A] [answered]
Hi all, 

I'm new to using pvlib. I'm right now using the single diode model and wanted to get 30 min instead of hourly generation data out of the model. Does this work or is pv lib only able to work with hourly data?

Thanks!

**Comment by AdamRJensen** [comment, marked as answer]
Hi @AnnaGo12 

There is no lower limit to the time resolution when using pvlib. In your case, you can probably simply pass in the input data with a higher temporal resolution (e.g., 30-minute) and it should work the same.

**Reply by AnnaGo12** [reply]
Thanks for the swift response Adam. I tried that, but I have trouble getting 30 min data for my typical meteorological year pvlib.iotools.get_pvgis_tmy(). It only gives me hourly data. Do you know how to fix that? I couldn't find details in the specifications.

**Reply by AdamRJensen** [reply]
TMY data is always hourly (it is a commonly used data format that attempts to approximate the long-term average conditions in a one-year dataset).

You could of course always interpolate the data to half-hourly. You should however be sure to be aware of the timestamp convention (is the data labeled by the left, center, or right edge, e.g., left labeled would correspond to 10 am being the average of the data points from 10 - 11 am).

15-minute data is available from CAMS (``pvlib.iotools.get_cams``) or the NSRDB ([``pvlib.iotools.get_psm3``](https://pvlib-python.readthedocs.io/en/v0.9.0/generated/pvlib.iotools.get_psm3.html)). Both of these datasets are free, but have different geographical coverage.

**Reply by AnnaGo12** [reply]
Thanks, that's very helpfu!

**Comment by mikofski** [comment]
[RELATED] would be great to address synthetically downscaling (or upsampling) inputs like this from hourly to subhourly - I think there is a related issue somewhere.

**Reply by kandersolar** [reply]
> I think there is a related issue somewhere.

https://github.com/pvlib/pvlib-python/issues/788, perhaps.  The topic is still of general interest to me, but I'm increasingly of the opinion that many of these papers don't describe their methods in sufficient detail for us to be able to implement them without having to make some assumptions.
